src/goImplementations.ts

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+import vscode = require('vscode');
+import cp = require('child_process');
+import path = require('path');
+import { byteOffsetAt, getBinPath, canonicalizeGOPATHPrefix } from './util';
+import { promptForMissingTool } from './goInstallTools';
+import { goKeywords, isPositionInString } from './util';
+import { getGoRuntimePath, resolvePath } from './goPath';
+
+interface GoListOutput {
+	Dir: string;
+	ImportPath: string;
+}
+
+interface GuruImplementsRef {
+	name: string;
+	pos: string;
+	kind: string;
+}
+
+interface GuruImplementsOutput {
+	type: GuruImplementsRef;
+	to: GuruImplementsRef[];
+	from: GuruImplementsRef[];
+}
+
+export class GoImplementationProvider implements vscode.ImplementationProvider {
+	public provideImplementation(document: vscode.TextDocument, position: vscode.Position, token: vscode.CancellationToken): Thenable<vscode.Definition> {
+		return new Promise<vscode.Definition>((resolve, reject) => {
+			if (token.isCancellationRequested) {
+				return resolve(null);
+			}
+
+			let listProcess = cp.execFile(getGoRuntimePath(), ['list', '-e', '-json'], { cwd: vscode.workspace.rootPath }, (err, stdout, stderr) => {
```

Comment:
Thanks for clarifying. I missed the `./...` you add later on to the `listOutput.ImportPath` and so the question